tests/test_data_processor.py

```
@@ -0,0 +1,42 @@
+def test_preprocess_data(data_processor):
+    data_processor.preprocess_data()
+    
+    assert data_processor.X.shape == (5, 4)
+    assert data_processor.y.shape == (5,)
+    assert set(data_processor.X.columns) == set(['num1', 'num2', 'cat1', 'cat2'])
+    assert data_processor.preprocessor is not None
+
+def test_split_data(data_processor):
+    data_processor.preprocess_data()
+    X_train, X_test, y_train, y_test = data_processor.split_data(test_size=0.4, random_state=42)
+    
```

Ruff

Blank line contains whitespace

Remove whitespace from blank line

(W293)